docs/modules/ROOT/pages/how-tos/configuring/build-with-args.adoc

```diff
@@ -0,0 +1,42 @@
+= Passing build arguments
+
+Konflux buildah task accepts link:https://github.com/konflux-ci/build-definitions/blob/main/task/buildah/0.1/buildah.yaml#L78[`BUILD_ARGS` parameter] that allows passing build arguments to the underlying `buildah` command via `--build-arg` flag. To do this, make the following changes to the PipelineRuns under the `.tekton/` directory:
+
+[source,diff]
+--
+     value: 'pip'
+   - name: build-source-image
+     value: 'true'
++  - name: build-args           (1)
++    value: ["ARG1=val", "ARG2=val"]
+   pipelineSpec:
+     finally:
+     - name: show-sbom
+@@ -127,6 +129,9 @@
+         during build
+       name: build-args-file
+       type: string
++    - name: build-args         (2)
++      type: array
++      default: []
+     results:
+     - description: ""
+       name: IMAGE_URL
+@@ -231,6 +236,8 @@
+         value: $(tasks.clone-repository.results.commit)
+       - name: BUILD_ARGS_FILE
+         value: $(params.build-args-file)
++      - name: BUILD_ARGS       (3)
++        value: ["$(params.build-args[*])"]
```

Comment:
It might also be good to show how to pass a dynamically generated value (e.g. an array result of a custom task), since that's the more useful use case for BUILD_ARGS

Reply:
Updated. PTAL whether that's what you meant.

Reply:
I think we could have a more complex example. Maybe add an "Advanced" section at the end, something like

### Advanced

You can do more creative things with the `BUILD_ARGS` parameter. For example, here's a snippet of a pipeline that allows passing build-args via the pipeline parameter (as described above) but also incorporates build-args generated dynamically by other tasks:

```yaml
spec:
  # ...
  pipelineSpec:
    tasks:
    # ...
    - name: build-container
      params:
      # ...
      - name: BUILD_ARGS
        value:
        # pass the build args array from the pipeline param
        - $(params.build-args[*])
        # and a dynamically generated array of build args
        - $(tasks.my-build-arg-generator-task.results.BUILD_ARGS[*])
        # and a build arg with a static name and dynamic value
        - VERSION=$(tasks.my-get-version-task.results.VERSION)
```